### PR TITLE
commit 11 get articles with topic query

### PR DESCRIPTION
**Pull request by CoupDeWhoop**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -127,26 +127,34 @@ describe('GET requests', () => {
                 expect(body.articles).toBeSorted({ key: "created_at", descending: true })
             })
         });
-        test('200 - should accept a topic query and respond with ', () => {
+        test('200 - articles endpoint should accept a topic query, filtering by specified value', () => {
```
Comment (pippaamy) [inline __tests__/app.test.js:133]: lets make a new test suite for this rather than changing previous tests

File: __tests__/app.test.js [hunk 2]
```diff
@@ -127,26 +127,34 @@ describe('GET requests', () => {
...
                     expect(article).toHaveProperty("article_img_url", expect.any(String))
                     expect(article).toHaveProperty("comment_count", expect.any(Number))
                 }) 
                 expect(body.articles).toBeSorted({ key: "created_at", descending: true })
             })
-        });
+        })
+        test('200 - topic query does not match any articles', () => {
+            return request(app)
+            .get("/api/articles?topic=trombone")
```
Comment (pippaamy) [inline __tests__/app.test.js]: if it doesnt match any it will be a 404

File: __tests__/app.test.js [hunk 3]
```diff
@@ -127,26 +127,34 @@ describe('GET requests', () => {
...
+            .expect(200)
+            .then(({ body }) => {
```
Comment (pippaamy) [inline __tests__/app.test.js:157]: what if it is a topic like topic=paper which exists but doesnt have any articles

**Review by pippaamy — COMMENTED** [review]
great job, some changes to make here! :)

File: __tests__/app.test.js (hunk 1 above)
Comment (CoupDeWhoop) [reply __tests__/app.test.js:133]: I think it looks like I'd changed a test but I'd just added them to my describe (GET /api/articles) endpoint. I've moveed them now into a new describe block.

Comment (pippaamy) [reply __tests__/app.test.js:133]: this one has been changed? when it should stay as /api/articles

File: __tests__/app.test.js [hunk 4]
```diff
@@ -127,26 +127,45 @@ describe('GET requests', () => {
...
                     expect(article).toHaveProperty("votes", expect.any(Number))
                     expect(article).toHaveProperty("article_img_url", expect.any(String))
                     expect(article).toHaveProperty("comment_count", expect.any(Number))
                 }) 
                 expect(body.articles).toBeSorted({ key: "created_at", descending: true })
             })
-        });
+        })
+        test("200 -  topic query exists but not does not appear in any articles", () => {
+            return request(app)
```
Comment (pippaamy) [inline __tests__/app.test.js:154]: nice!

**Review by pippaamy — COMMENTED** [review]
ello, good work changes those tests! We don't want to be changing previous tests - but all good to merge

**Pull request merged**